src/scale.js

```diff
@@ -0,0 +1,81 @@
+/* global WebAnimationsPolyfillExtension internalScope */
+
+(function () {
+  function isNumeric (number) {
+    return !isNaN(number);
+  }
+
+  function addDefaultAxis (values) {
+    if (values.length === 1) { // x-axis was specified
+      values.push(1, 1);
+    } else if (values.length === 2) { // x and y-axis were specified
+      values.push(1);
+    }
+    return values;
+  }
+
+  function scaleParse (input) {
+    /* According to spec:
+      https://drafts.csswg.org/css-transforms-2/#propdef-scale
+      unspecified scales default to 1
+    */
+    if (input === undefined) {
+      return null;
+    } else if (input === 'none') {
+      return [1, 1, 1];
+    }
+
+    var values = input.split(/\s+/);
+    var numValues = values.length;
+    if (numValues < 1 || numValues > 3) {
+      return undefined;
+    }
+
+    for (var i = 0; i < numValues; i++) {
+      if (values[i] === '') {
+        return undefined;
+      }
+
+      if (!isNumeric(values[i])) {
+        return undefined;
+      }
+    }
+
+    for (var j = 0; j < numValues; j++) {
+      values[j] = Number(values[j]);
+    }
+
+    values = addDefaultAxis(values);
```

Comment:
Don't have this function return the input if it's going to mutate it. It's misleading at the callsite.